File: javascript/007/main.js
/*
 * ProjectEuler 007
 * JavaScript / main.js
 * copyright (c) 2015 Susisu
 *
 * Node.js v0.12.0
 * `node main.js`
 */

function main() {
    var count = 0;
    var n = 2;
    while (true) {
        if (isPrime(n)) {
            count++;
            if (count === 10001) {
                console.log(n);
                break;
            }
        }
        n++;
    }
}

function isPrime(n) {
    if (n <= 1) {
        return false;
    }
    else {
        var r = Math.sqrt(n);
        for (var m = 2; m <= r; m++) {
            if (n % m === 0) {
                return false;
            }
        }
        return true;
    }
}

main();

File: go/007/main.go
/*
 * ProjectEuler 007
 * Go / main.go
 * copyright (c) 2015 Susisu
 *
 * Go 1.4.1
 * `go build -o bin/main main.go`
 */


package main

import (
    "fmt"
    "math"
)

func main() {
    counter := 0
    n := 2
    for true {
        if isPrime(n) {
            counter++
            if counter == 10001 {
                break
            }
        }
        n++
    }
    fmt.Printf("%d\n", n)
}

func isPrime(n int) bool {
    if n < 2 {
        return false
    }
    r := int(math.Sqrt(float64(n)));
    for i := 2; i <= r; i++ {
        if n % i == 0 {
            return false
        }
    }
    return true
}
